caracal/workers/selfcal_worker.py

```diff
@@ -282,7 +283,19 @@ def worker(pipeline, recipe, config):
             freq_chunk = int(max(all_freq_solution))
 
     min_uvw = config['minuvw_m']
+
     ncpu = config['ncpu']
+    if ncpu == 0:
+      ncpu = psutil.cpu_count()
+    else:
+      ncpu = min(ncpu, psutil.cpu_count())
+
+    nrdeconvsubimg = ncpu if config['img_nrdeconvsubimg'] == 0 else config['img_nrdeconvsubimg']
+    if nrdeconvsubimg == 1:
+        wscl_parallel_deconv = None
+    else:
+        wscl_parallel_deconv = np.ceil(config['img_npix']/np.sqrt(nrdeconvsubimg)).astype(int)
```

Comment:
Please change astype(int) to overall int(np.ceil....) instead, since astype(int) seems to give int64 type instead of int and that's the wrong datatype for parallel-deconvolution.